javascript/phone-number/phone-number.js

```diff
@@ -0,0 +1,39 @@
+var PhoneNumber = function(input){
+	//validate the input type
+	if(typeof input !== "string") {
+		throw "Invalid input type. Please give a number in string and try again.";
+	}
+	this.num = input;
+};
+
+PhoneNumber.prototype.number = function(){
+	//clear the non-number char in the string
+	var regex = /[0-9]/g;
+	var n = this.num.match(regex).join("");
```

Comment:
An alternative, more terse approach might be removing all non-digits with `String.prototype.replace` and `\D`.